src/Agent.Worker/TaskRunner.cs

```diff
@@ -209,6 +209,19 @@ public async Task RunAsync()
                 // Expand the inputs.
                 Trace.Verbose("Expanding inputs.");
                 runtimeVariables.ExpandValues(target: inputs);
+
+                if (Task.Name.StartsWith("__system_posttargettask_")
+                    || Task.Name.StartsWith("__system_pretargettask_"))
+                {
+                    bool isValid = this.ValidateInjectedTaskInputs(inputs);
+                    if (!isValid)
+                    {
+                        ExecutionContext.Result = TaskResult.Skipped;
+                        ExecutionContext.ResultCode = $"It is not allowed to pass inputs that contain secrets to the tasks injected by decorators";
```

Comment:
Can we share target task here for which it failed?

Reply:
Added logic for printing input names that contains secrets